tests/dotnet/UnitTests/ProjectTest.cs

```diff
@@ -362,5 +362,36 @@ IEnumerable<string> FilterToAssembly (IEnumerable<string> lines, string assembly
 					return true;
 				});
 		}
+
+		// This is copied from the KillEverything method in xharness/Microsoft.DotNet.XHarness.iOS.Shared/Hardware/SimulatorDevice.cs and modified to work here.
```

Comment:
unclear why it's only in DotNet tests ?

and if because it's in xharness then why does not it solve the problem for dotnet ?

Reply:
I don't know the exact conditions when this happens on the bots 😒 

I know that this problem went away in another PR when I ran _only_ the dotnet tests.

I can also reproduce locally by running `killall -9 "iPhone Simulator" "iOS Simulator" Simulator "Simulator (Watch)" com.apple.CoreSimulator.CoreSimulatorService` before running the unit tests. xharness also calls `killall` like this (before running tests in a specific simulator), so it might be an ordering issue where if xharness does stuff in a certain order then the problem occurs (it does not happen if I do a `make runner` and then execute the dotnet tests manually). Another potential fix might be to add the `simctl list` command to xharness (since that's what actually seems to fix the problem) - and hope that works.

Reply:
ah, ok. I just wondered if this should be part of more (all?) unit tests app we run - instead of in xharness.